config.wyam

```diff
@@ -845,6 +852,32 @@ public class FeedItemData
     }
 }
 
+public IEnumerable<IDocument> GetOpmlData(IReadOnlyList<IDocument> inputs, IExecutionContext context)
+{
+    return inputs.Select(
+        input => {
+            string feed = input.String("Feed");
+            string title = input.String("Title");
+
+            if (string.IsNullOrWhiteSpace(title)) {
+                // process the base Url as the title in the absence of one
+                var slashes = "//";
+                
+                int from = feed.IndexOf(slashes) + slashes.Length;
+                int to = feed.IndexOf("/", from);
+
+                title = feed.Substring(from, to - from);
+            }
+
+            MetadataItems metadata = new MetadataItems(){
+                { "Title", title },
+                { "Feed", feed }
+            };
+
+            return context.GetDocument(input, metadata);

```

Comment:
Creating a new document is a little expensive (the engine caches documents on creation - one of the reasons we have to use a factory to get them). Can we move this inside the `if (string.IsNullOrWhiteSpace(title))` and return the original document (`return input`) if the title isn't changed?